language/functional_tests/tests/testsuite/transaction_fee_distribution/test_txn_fee_dust_distribution.mvir

```diff
@@ -0,0 +1,25 @@
+// The fees collected is less than the number of validators
+
+//! account: vivian, 1000000, 0, validator
+//! account: valentina, 1000000, 0, validator
+//! sender: vivian
+
+import 0x0.LibraSystem;
+import 0x0.LibraAccount;
+import 0x0.LibraCoin;
+import 0x0.Vector;
+
+main() {
+  let lib_coin: LibraCoin.T;
+  let number_of_validators: u64;
+
+  number_of_validators = LibraSystem.validator_set_size();
+  assert(copy(number_of_validators) > 1, 0);
+  lib_coin = LibraAccount.withdraw_from_sender(copy(number_of_validators) - 1);
+
+  LibraAccount.deposit({{fees}} , move(lib_coin));
+  LibraSystem.block_prologue(1, h"aa", h"bb", {{vivian}});
```

Comment:
Nit: could you call this is a separate transaction? In the not-too-distant future, we'll hopefully make `block_prologue` private and add a `//! new-block` directive that lets us test things like this.

Reply:
Calling it in a separate transaction could be done but it would be more work -- the transaction fee from the first transaction would make the pot of fees greater than the number of validators (unless we have a lot of validators defined). In order to do this, I'll probably need to add the gas price directive to the functional tests (which probably isn't that bad of an idea...)